### ISSUE TITLE
Proj prints out the batch code on CMD each time I execute a conda command

### ISSUE THREAD

**Issue by l3d00m** [conda-forge/proj.4-feedstock#79]
Issue: Each time I execute a successful conda command, it then prints the following code **on the CMD**. This is an examplary CMD output, I only entered the conda install command, rest was automatically:

```
(this_env) C:\>conda install qt
Collecting package metadata (current_repodata.json): done
Solving environment: done

# All requested packages already installed.

C:\>if exist C:\Users\myuser\.conda\envs\this_env\Library\share\proj\copyright_and_licenses.csv (
rem proj-data is installed because its license was copied over

)  else ()

(this_env) C:\>
```
This was first reported in https://github.com/conda/conda/issues/9838. I was able to trace the content back to [this file](https://github.com/conda-forge/proj.4-feedstock/blob/master/recipe/scripts/activate.bat), which is why I decided to open this issue here too. For me it seems like a conda bug as well, because it should never print code on the cmd?

This did occur after updating to proj version 7.

<br/>
Environment (<code>conda list</code>):
<details>

```
$ conda list
# Name                    Version                   Build  Channel
# Sorry, I had to censor a lot of packages because there were some company internal packages in there and I don't want to be doxxed
numpy                     1.18.1           py38ha749109_1    conda-forge
openjpeg                  2.3.1                h57dd2e7_3    conda-forge
openssl                   1.1.1f               hfa6e2cd_0    conda-forge
packaging                 20.1                       py_0    conda-forge
pandas                    1.0.3            py38hbe40bda_0    conda-forge
pcre                      8.44                 h6538335_0    conda-forge
pip                       20.0.2                     py_2    conda-forge
proj                      7.0.0                hfe17d32_2    conda-forge
pycodestyle               2.5.0                      py_0    conda-forge
pycparser                 2.20                       py_0    conda-forge
pygments                  2.6.1                      py_0    conda-forge
pylint                    2.4.4                    py38_0    conda-forge
pyopenssl                 19.1.0                     py_1    conda-forge
pyparsing                 2.4.7              pyh9f0ad1d_0    conda-forge
pyproj                    2.6.0            py38h1d58662_1    conda-forge
pyqt                      5.12.3           py38h6538335_1    conda-forge
pyqt5-sip                 4.19.18                  pypi_0    pypi
pyqtwebengine             5.12.1                   pypi_0    pypi
pysocks                   1.7.1            py38h32f6830_1    conda-forge
python                    3.8.2           h5fd99cc_5_cpython    conda-forge
python-dateutil           2.8.1                      py_0    conda-forge
pytz                      2019.3                     py_0    conda-forge
qt                        5.12.5               h7ef1ec2_0    conda-forge
requests                  2.23.0             pyh8c360ce_2    conda-forge
rope                      0.16.0                     py_0    conda-forge
rtree                     0.9.4            py38h7ad75cc_1    conda-forge
scipy                     1.3.2            py38h582fac2_0    conda-forge
setuptools                46.1.3           py38h32f6830_0    conda-forge
# And censored again, i think only the proj and python version should be relevant
```
</details>

<br/>
Details about  <code>conda</code> and system ( <code>conda info</code> ):
<details>

```
$ conda info
# Again had to remove some lines, sorry
          conda version : 4.7.12
    conda-build version : 3.18.8
         python version : 3.7.4.final.0
       virtual packages :
               platform : win-64
             user-agent : conda/4.7.12 requests/2.22.0 CPython/3.7.4 Windows/10 Windows/10.0.17763
          administrator : False
             netrc file : None
           offline mode : False
```
</details>

**Comment by xylar** [comment]
@l3d00m, sorry about that!  I'm not very good at batch files.

It looks to me like `if` should have been `@if`.  could you manually make that change in the version of that file in your local environment, then activate the environment and let me know if that fixes the problem for you?  If so, I can make a new build with the fix.

**Comment by xylar** [comment]
Hmm, on further investigation, I don't think that's it.  I'll see what I can figure out.

**Comment by l3d00m** [comment]
Hi @xylar I added the @ to the if and it seems to be fixed for me, will try in more detail later though

**Comment by xylar** [comment]
Okay, I think it's not anything `conda` is doing and you should close the issue there.  Without the `@` sign, it's just that the batch script is verbose and outputs each command as it runs.  (What's wrong with you, Microsoft!?)

**Comment by xylar** [comment]
I'll put in a fix, but the existing behavior is harmless if annoying.

**Comment by l3d00m** [comment]
Thanks for the quick fix. I left a comment in the conda issue, as I'm not the author. Noone likes batch scripts I guess 😛

**Comment by xylar** [comment]
@DrMeshugge [wrote](https://github.com/conda/conda/issues/9838#issuecomment-622317639):
> Hey, I ran into the same issue after upgrading to jupyterlab 2.1.1; specifically after running:
```
conda update -c conda-forge jupyterlab
conda upgrade jupyterlab jupyterlab_server
```
> When I open Anaconda Prompt:
```
C:\Users\myuser>if exist C:\Users\myuser\Anaconda3\Library\share\proj\copyright_and_licenses.csv ( rem proj-data is installed because its license was copied over ) else ()
```
> It seems the upgrade broke my Anaconda Navigator (doesn’t open anymore) and most conda commands lead to errors. There is no copyright_and_licenses.csv in the proj folder.

**Comment by xylar** [comment]
@DrMeshugge, this should have been fixed by the latest build of `proj` 7.0.0 (build 3) that I did 2 weeks ago.  Can you confirm with `conda list` that you have that build and not an earlier one?
``` bash
$ conda list proj
```
If you have an older version, can you try to update?  This should do it:
```
$ conda install -c conda-forge proj
```

**Comment by DrMeshugge** [comment]
Hi @xylar , I have `proj` 7.0.0 (hf5a9257_1) running on my system. But I cannot run `$ conda install` as most conda prompts result in error messages for several files in:

``` 
C:\Users\myuser\Anaconda3\lib\site-packages\conda\...
and
C:\Users\myuser\Anaconda3\lib\site-packages\urllib3\...
```

**Comment by xylar** [comment]
@DrMeshugge, I'm sorry to hear that.  I would need to see the details of the error messages to be of more help.

Are you able to run `conda create` to make a new environment?  If so, that might be the better option, e.g.:
```
conda create -n test -c cond-forge proj=7.0.0
```

I don't think it's related, but you will want to follow the instructions at https://conda-forge.org/:
```
conda config --add channels conda-forge
conda config --set channel_priority strict
```
to make sure you are getting packages from `conda-forge` wherever possible to avoid conflicting versions between different channels.

**Comment by DrMeshugge** [comment]
Unfortunately, I cannot even create new environments anymore. I’m receiving following error messages when executing conda prompts:

`$conda create -n test -c cond-forge proj=7.0.0`
```
Traceback (most recent call last):

  File "C:\Users\myuser\Anaconda3\lib\site-packages\conda\exceptions.py", line 1079, in __call__

    return func(*args, **kwargs)

  File "C:\Users\myuser\Anaconda3\lib\site-packages\conda\cli\main.py", line 84, in _main

    exit_code = do_call(args, p)

  File "C:\Users\myuser\Anaconda3\lib\site-packages\conda\cli\conda_argparse.py", line 80, in do_call

    module = import_module(relative_mod, __name__.rsplit('.', 1)[0])

  File "C:\Users\myuser\Anaconda3\lib\importlib\__init__.py", line 127, in import_module

    return _bootstrap._gcd_import(name[level:], package, level)

  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import

  File "<frozen importlib._bootstrap>", line 983, in _find_and_load

  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked

  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked

  File "<frozen importlib._bootstrap_external>", line 728, in exec_module

  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed

  File "C:\Users\myuser\Anaconda3\lib\site-packages\conda\cli\main_create.py", line 10, in <module>

    from .install import install

  File "C:\Users\myuser\Anaconda3\lib\site-packages\conda\cli\install.py", line 19, in <module>

    from ..core.index import calculate_channel_urls, get_index

  File "C:\Users\myuser\Anaconda3\lib\site-packages\conda\core\index.py", line 10, in <module>

    from .package_cache_data import PackageCacheData

  File "C:\Users\myuser\Anaconda3\lib\site-packages\conda\core\package_cache_data.py", line 15, in <module>

    from .path_actions import CacheUrlAction, ExtractPackageAction

  File "C:\Users\myuser\Anaconda3\lib\site-packages\conda\core\path_actions.py", line 30, in <module>

    from ..gateways.connection.download import download

  File "C:\Users\myuser\Anaconda3\lib\site-packages\conda\gateways\connection\__init__.py", line 21, in <module>

    from requests import ConnectionError, HTTPError, Session

  File "C:\Users\myuser\Anaconda3\lib\site-packages\requests\__init__.py", line 43, in <module>

    import urllib3

  File "C:\Users\myuser\Anaconda3\lib\site-packages\urllib3\__init__.py", line 7, in <module>

    from .connectionpool import HTTPConnectionPool, HTTPSConnectionPool, connection_from_url

  File "C:\Users\myuser\Anaconda3\lib\site-packages\urllib3\connectionpool.py", line 39, in <module>

    from .response import HTTPResponse

  File "C:\Users\myuser\Anaconda3\lib\site-packages\urllib3\response.py", line 155, in <module>

    class HTTPResponse(io.IOBase):

  File "C:\Users\myuser\Anaconda3\lib\site-packages\urllib3\response.py", line 377, in HTTPResponse

    DECODER_ERROR_CLASSES += (brotli.error,)

AttributeError: module 'brotli' has no attribute 'error'



During handling of the above exception, another exception occurred:



Traceback (most recent call last):

  File "C:\Users\myuser\Anaconda3\Scripts\conda-script.py", line 12, in <module>

    sys.exit(main())

  File "C:\Users\myuser\Anaconda3\lib\site-packages\conda\cli\main.py", line 150, in main

    return conda_exception_handler(_main, *args, **kwargs)

  File "C:\Users\myuser\Anaconda3\lib\site-packages\conda\exceptions.py", line 1371, in conda_exception_handler

    return_value = exception_handler(func, *args, **kwargs)

  File "C:\Users\myuser\Anaconda3\lib\site-packages\conda\exceptions.py", line 1082, in __call__

    return self.handle_exception(exc_val, exc_tb)

  File "C:\Users\myuser\Anaconda3\lib\site-packages\conda\exceptions.py", line 1126, in handle_exception

    return self.handle_unexpected_exception(exc_val, exc_tb)

  File "C:\Users\myuser\Anaconda3\lib\site-packages\conda\exceptions.py", line 1137, in handle_unexpected_exception

    self.print_unexpected_error_report(error_report)

  File "C:\Users\myuser\Anaconda3\lib\site-packages\conda\exceptions.py", line 1207, in print_unexpected_error_report

    from .cli.main_info import get_env_vars_str, get_main_info_str

  File "C:\Users\myuser\Anaconda3\lib\site-packages\conda\cli\main_info.py", line 19, in <module>

    from ..core.index import _supplement_index_with_system

  File "C:\Users\myuser\Anaconda3\lib\site-packages\conda\core\index.py", line 10, in <module>

    from .package_cache_data import PackageCacheData

  File "C:\Users\myuser\Anaconda3\lib\site-packages\conda\core\package_cache_data.py", line 15, in <module>

    from .path_actions import CacheUrlAction, ExtractPackageAction

  File "C:\Users\myuser\Anaconda3\lib\site-packages\conda\core\path_actions.py", line 30, in <module>

    from ..gateways.connection.download import download

  File "C:\Users\myuser\Anaconda3\lib\site-packages\conda\gateways\connection\__init__.py", line 21, in <module>

    from requests import ConnectionError, HTTPError, Session

  File "C:\Users\myuser\Anaconda3\lib\site-packages\requests\__init__.py", line 43, in <module>

    import urllib3

  File "C:\Users\myuser\Anaconda3\lib\site-packages\urllib3\__init__.py", line 7, in <module>

    from .connectionpool import HTTPConnectionPool, HTTPSConnectionPool, connection_from_url

  File "C:\Users\myuser\Anaconda3\lib\site-packages\urllib3\connectionpool.py", line 39, in <module>

    from .response import HTTPResponse

  File "C:\Users\myuser\Anaconda3\lib\site-packages\urllib3\response.py", line 155, in <module>

    class HTTPResponse(io.IOBase):

  File "C:\Users\myuser\Anaconda3\lib\site-packages\urllib3\response.py", line 377, in HTTPResponse

    DECODER_ERROR_CLASSES += (brotli.error,)

AttributeError: module 'brotli' has no attribute 'error'
```

**Comment by xylar** [comment]
Thank you for the details.  I looks like your base environment may have been corrupted in some way with incompatible packages.  I'm afraid you may not have an alternative to installing Anaconda again from scratch.  I don't think this issue is related to `proj`, at least I don't see an obvious reason it would be, but I can't be sure.

Depending on your workflow, you might be better off installing miniconda instead of anaconda.  Miniconda would let you have a small, light-weight `base` environment from which you can create one or more "disposable" environments.  If you do this and one of your environments other than `base` gets corrupted somehow, you can just delete it and reinstall the packages you need.

Obviously, you're not having a very fun experience with anaconda, and I'm sorry that the bug I introduced to activating of the `proj` package has contributed to that.  Hopefully, a fresh start with either anaconda or miniconda will also give you a better experience.  I wish I could recommend a way to fix your existing installation but it does look like it might be messed up beyond repair.

**Comment by DrMeshugge** [comment]
Thank you @xylar for looking into this. I will try to keep my environments more organized in the future – there seem to be good reasons for doing so. 😉

**Comment by mwight07** [comment]
For me the issue was fixed by finding the proj4-activate.bat file in the activate.d folder and adding the @ in front of the if. You may also have to change it in the activate.bat file. I found both in the following paths:
C:\Anaconda\pkgs\proj-7.0.0-hfe17d32_2\info\recipe\scripts\activate.bat
C:\Anaconda\pkgs\proj-7.0.0-hfe17d32_2\etc\conda\activate.d\proj4-activate.bat

**Comment by vempi** [comment]
> @DrMeshugge, this should have been fixed by the latest build of `proj` 7.0.0 (build 3) that I did 2 weeks ago. Can you confirm with `conda list` that you have that build and not an earlier one?
> 
> ```shell
> $ conda list proj
> ```
> 
> If you have an older version, can you try to update? This should do it:
> 
> ```
> $ conda install -c conda-forge proj
> ```

This works for me! Thanks!
